src/components/homepage/Homepage.tsx

```diff
@@ -1,45 +1,77 @@
-import React from 'react';
-import { useSelector, useDispatch } from 'react-redux';
-import { RootState } from '../../state/store';
-import { selectResource } from '../../state/actions/resourceActions';
-import { Link } from 'react-router-dom';
-import { DataFetched } from '../../state/reducers/resource';
-import { requestResource } from '../../state/thunkMiddleware';
+import React from "react";
+import { useSelector, useDispatch } from "react-redux";
+import { RootState } from "../../state/store";
+import { selectResource } from "../../state/actions/resourceActions";
+import { Link } from "react-router-dom";
+import { DataFetched } from "../../state/reducers/resource";
+import { requestResource } from "../../state/thunkMiddleware";
+import {
+  Card,
+  CardActionArea,
+  Typography,
+  Grid,
+  CircularProgress,
+} from "@material-ui/core";
+import useStyles from "./style";
+import { ReactComponent as FhirLogo } from "../../assets/img/fhir-logo.svg";
+import NavBar from "./navBar/NavBar";
 
 const Homepage: React.FC<{}> = () => {
-    const { loading, data } = useSelector((state: RootState) => state.resource);
-    const dispatch = useDispatch();
-    const dispatchResourceSelected = (data: DataFetched): void => {
-        if (data.id) {
-            dispatch(selectResource(data.id));
-            dispatch(requestResource(data.id));
-        };
-    };
+  const { loading, data } = useSelector((state: RootState) => state.resource);
+  const dispatch = useDispatch();
+  const classes = useStyles();
 
-    const mapAllResources = data?.map((data: DataFetched, index: number) => {
-        return (
-            <Link to="/editprofile" key={index}>
-                <button onClick={
-                    (): void => {
-                        dispatchResourceSelected(data);
-                    }
-                }>
-                    {data.id}
-                </button>
-            </Link>
-        );
-    })
+  const dispatchResourceSelected = (data: DataFetched): void => {
+    if (data.id) {
+      dispatch(selectResource(data.id));
+      dispatch(requestResource(data.id));
+    }
+  };
 
-    if (loading === true) {
-        return <p>Loading...</p>
-    };
+  const mapAllResources = data?.map((data: DataFetched, index: number) => {
```

Comment:
Careful when you use shadow variables. data is used twice for different variables